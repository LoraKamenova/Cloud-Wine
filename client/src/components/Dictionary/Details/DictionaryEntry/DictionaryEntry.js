import './DictionaryEntry.css'

const DictionaryEntry = (props) => {
    return (
        <article className="dictionary-article">
            <div className="dictionary-article-header" style={{backgroundImage: `url(${props.imageUrl})`}}>
                <span className="dictionary-article-heading">{props.title}</span>
            </div>
            <p className="dictionary-article-paragraph">{props.fullText}</p>
        </article>
    );
};

export default DictionaryEntry;
