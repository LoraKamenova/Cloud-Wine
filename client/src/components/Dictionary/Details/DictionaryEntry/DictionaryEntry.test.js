import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import DictionaryEntry from './DictionaryEntry';

describe('DictionaryEntry Component', () => {
    it('Should display entry title', () => {
        render(
            <BrowserRouter>
                <DictionaryEntry title="Аериране" />
            </BrowserRouter>
        );

        expect(document.querySelector(".dictionary-article-heading").textContent).toBe('Аериране');
    });

    it('Should display entry content', () => {
        render(
            <BrowserRouter>
                <DictionaryEntry fullText="Аерирането на виното означава излагане на виното на въздух или с други думи оставянето му да „подиша“ преди да бъде изпито." />
            </BrowserRouter>
        );

        expect(document.querySelector(".dictionary-article-paragraph").textContent).toBe('Аерирането на виното означава излагане на виното на въздух или с други думи оставянето му да „подиша“ преди да бъде изпито.');
    });

});
