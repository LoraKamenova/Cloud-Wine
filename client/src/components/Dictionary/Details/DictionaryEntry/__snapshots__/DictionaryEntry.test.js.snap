// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictionaryEntry Component DictionaryEntry Component snapshot 1`] = `
<DocumentFragment>
  <article
    class="dictionary-article"
  >
    <div
      class="dictionary-article-header"
      style="background-image: url(undefined);"
    >
      <span
        class="dictionary-article-heading"
      />
    </div>
    <p
      class="dictionary-article-paragraph"
    />
  </article>
</DocumentFragment>
`;
